jwst/regtest/st_fitsdiff.py

```diff
@@ -1196,113 +1221,132 @@ def _diff(self):
-                    if idx == last_seen_idx:
-                        # Skip duplicate indices, which my occur when the column
-                        # data contains multi-dimensional values; we're only
-                        # interested in storing row-by-row differences
-                        continue
-                    last_seen_idx = idx
-                    self.diff_values.append(((col.name, idx), (arra[idx], arrb[idx])))
-
-            else:
-                # The following lines include STScI's changes:
-                # - Calculate the absolute and relative differences separately for ad hoc report
-                # - If report_pixel_loc_diffs is False, just get the total differences,
-                #   it is not important where they come from
-
-                # Calculate the absolute and relative differences
-                get_stats = False
-                if np.issubdtype(arra.dtype, np.floating) and np.issubdtype(
-                    arrb.dtype, np.floating
-                ):
                     nan_idx = np.isnan(arra) | np.isnan(arrb)
                     anonan = arra[~nan_idx]
                     bnonan = arrb[~nan_idx]
                     with warnings.catch_warnings():
                         warnings.simplefilter("ignore", RuntimeWarning)
-                        diffs = np.abs(anonan - bnonan)
+                        diffs = np.abs(bnonan - anonan)
                     abs_diffs = diffs[diffs > self.atol].size
                     nozeros = (diffs != 0.0) & (bnonan != 0.0)
                     rel_values = diffs[nozeros] / np.abs(bnonan[nozeros])
                     rel_diffs = rel_values[rel_values > self.rtol].size
                     get_stats = True
 
-                elif "P" in col.format or "Q" in col.format:
-                    diffs = []
-                    rel_values = []
-                    zeros_idx = []
-                    nan_idx = []
-                    for idx in range(len(arra)):
-                        if arra[idx] == 0 or arrb[idx] == 0:
-                            zeros_idx.append(idx)
-                        elif np.isnan(arra[idx]) or np.isnan(arrb[idx]):
-                            nan_idx.append(idx)
-                        else:
-                            df = np.abs(arra[idx] - arrb[idx])
-                            if not df <= self.atol:
-                                diffs.append(df)
-                            dfr = df / np.abs(arrb[idx])
-                            if not dfr <= self.rtol:
-                                rel_values.append(dfr)
-                    get_stats = True
-                    diffs = np.array(diffs)
-                    abs_diffs = diffs.size
-                    rel_values = np.array(rel_values)
-                    rel_diffs = rel_values.size
-                    anonan = arra[~np.array(nan_idx)]
-                    bnonan = arrb[~np.array(nan_idx)]
-
+            elif "P" in col.format or "Q" in col.format:
```

Comment:
The file /grp/crds/cache/references/hst/51c1638pi_imp.fits is an example of a FITS table with P formatting, allowing testing of this block.  When I tried it, I got this error:
`File ~/jwst/git/jwst/jwst/regtest/st_fitsdiff.py:1317, in STTableDataDiff._diff(self)
   1315 nan_idx = []
   1316 for idx in range(len(arra)):
-> 1317     if arra[idx] == 0 or arrb[idx] == 0:
   1318         zeros_idx.append(idx)
   1319     elif np.isnan(arra[idx]) or np.isnan(arrb[idx]):

ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`

Reply:
thanks, Robert. I fixed the issue